Source/misdat.cpp

```diff
@@ -127,68 +129,162 @@ MissileDataStruct MissileData[] = {
-	{ "ex_blu3",  MFILE_EXBL3,       1, 1, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, {  7,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 }, { 292,   0,   0,   0,   0,   0,   0,   0,  0,  0,  0,  0,  0,  0,  0,  0 }, { 114,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 } },
-	{ "",         MFILE_NONE,        0, 0, { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr }, { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, {  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 }, {   0,   0,   0,   0,   0,   0,   0,   0,  0,  0,  0,  0,  0,  0,  0,  0 }, {   0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0 } },
+	// name,      animName,        animFAmt, flags,                animDelay[16], animLen[16],                           animWidth[16],          animWidth2[16]
+	{ "Arrows",   MFILE_ARROWS,     1,       MFLAG_LOCK_ANIMATION,    0,            16,                                     96,                     16               },
+	{ "Fireba",   MFILE_FIREBA,    16,       0,                       0,            14,                                     96,                     16               },
+	{ "Guard",    MFILE_GUARD,      3,       0,                       1,          { 15, 14,  3 },                           96,                     16               },
+	{ "Lghning",  MFILE_LGHNING,    1,       0,                       0,             8,                                     96,                     16               },
+	{ "Firewal",  MFILE_FIREWAL,    2,       0,                       0,          { 13, 11 },                              128,                     32               },
+	{ "MagBlos",  MFILE_MAGBLOS,    1,       0,                       1,            10,                                    128,                     32               },
+	{ "Portal",   MFILE_PORTAL,     2,       0,                    {  0, 1 },       16,                                     96,                     16               },
+	{ "Bluexfr",  MFILE_BLUEXFR,    1,       0,                       0,            19,                                    160,                     48               },
+	{ "Bluexbk",  MFILE_BLUEXBK,    1,       0,                       0,            19,                                    160,                     48               },
+	{ "Manashld", MFILE_MANASHLD,   1,       MFLAG_LOCK_ANIMATION,    0,             1,                                     96,                     16               },
+	{ "Blood",    MFILE_BLOOD,      4,       0,                       0,          { 15,  8,  8,  8 },                    {  96, 128, 128, 128 }, {  16, 32, 32, 32 } },

```

Comment:
Blood is unused, since it is the only one that doesn't use the same value for all entries in `animWidth` and `animWidth2` we could blank it and that would allow use to change the type from `std::array<int16_t, 16>` to simply `int16_t` which would achieve much of the simplification done here but with out the need for the additional constructors.